echopop/survey.py

```diff
@@ -658,6 +658,148 @@ def kriging_analysis(
+
+        axis_limits: Optional[Dict[str, Any]]
+
+            A dictionary that contains limits for the x- and y-axes. This should contain two
+            nested dictionaries for `x` and `y`:
+
+            - `xmin`/`ymin` (float): Minimum x- and y-axis values.
+            - `xmax`/`ymax` (float): Maximum x- and y-axis values.
+            - `left`/`right` (float): The left- and right-most axis values. This should not be
+            included if `xmin`/`xmax`/`ymin`/`ymax` are included.
+
+        geo_config: Optional[Dict[str, Any]]
+            A dictionary that contains three keyword arguments:
+
+            - `coastline` (`cartopy.feature.Feature`): A `cartopy.feature.Feature` object that
+            includes land and/or coastline information used for geospatial plots.
+
+            - `init` (str): The initial geospatial projection code (e.g. "EPSG:4326").
+
+            - `plot_projection` (`cartopy.crs.Projection`): A `cartopy.crs.Projection`
+            projection object used for plotting geospatial data.
+
+        grid_heatmap: Optional[bool]
+            An optional boolean value for adding correctly spaced gridlines to the biological
+            heatmap. It is otherwise not used for the spatial plots.
+        sex: Optional[Literal["all", "female", "male"]]
+            An optional argument for defining which fish sex to plot for the biological
+            heatmap. It is otherwise not used for the spatial plots.
+        log_base: Optional[float]
+            Base for rescaling the plot colormap via logarithmic transformation.
+        cmap: Optional[str]
+            Plotting colormap (e.g. "viridis").
+        vmin, vmax: Optional[float]
+            The data range used to rescale the colormap in linear or logarithmic space.
+        **kwargs: Dict[str, Any]
+            Plotting functions can accept additional arguments used within
+            :func:`matplotlib.pyplot.plot`
+
+        See Also
+        ----------------
+        :func:`matplotlib.pyplot.plot`
+            For more details on `matplotlib` plotting function keyword arguments. Keep in mind that
+            these should be used in caution as they may interfere with the primary variables
+            defined for `plotting_parameters`.
+
+        """
+
+        # Get associated plotting function information
+        plot_info = egp.PLOT_MAP(self, kind)
+
+        # Initialize 'parameters' dictionary
+        parameters = plot_parameters.copy()
+
+        # Proceed with plotting
+        # ---- Type: spatial
+        if plot_info["type"] == "spatial":
+            # ---- Get the geospatial configuration
+            geo_config = self.config["geospatial"].copy()
+            # ---- Get the coastline, if it exists, from `plot_parameters`
+            if parameters.get("geo_config"):
+                geo_config.update(parameters.get("geo_config", None))
+            # ---- Update the parameterization
+            parameters.update(dict(geo_config=geo_config))
```

Comment:
A few comments:
- do you need to define a new variable `plot_parameters`? 
- there is a comment `"Get the coastline, if it exists, from plot_parameters"`, but the operation would update the `parameter` dict with _everything_ in `geo_config`, not just coastline
- the if check for `parameters.get("geo_config")` is confusing: when `parameters["geo_config"]` does not exist, the statement `geo_config.update(parameters.get("geo_config", None))` will not be executed, but in there you have in there to update the dict `geo_config` with None. I am not following what you are trying to achieve here, but I think there's some redundancy here that can be removed.

Reply:
> do you need to define a new variable plot_parameters?

This was done to keep things a bit tidier and to streamline how plotting parameters are fed into the `PlotModel` Pydantic validator. One alternative would be to keep `plot_parameters` for `Echopop`-specific plotting parameters and allow for `**kwargs` to include whichever `matplotlib` keyword arguments.

> there is a comment "Get the coastline, if it exists, from plot_parameters", but the operation would update the parameter dict with everything in geo_config, not just coastline

The purpose of this chunk is to pull in geospatial parameterization stored within the `Survey` object (primarily the projection, which in this case provides a consistent usage of the same projection between spatial calculations and plotting).

```python
from echopop.survey import Survey

# Mock
self = Survey
self.config = {"geospatial": {"init": "epsg:4326"}}
parameters = {"vmin": 0, "vmax": 100, "geo_config": {}}
```

This yields: `{'vmin': 0, 'vmax': 100, 'geo_config': {'init': 'epsg:4326'}}`

```python

# Mock
self = Survey
self.config = {"geospatial": {"init": "epsg:4326"}}
parameters = {"vmin": 0, "vmax": 100, "geo_config": {"init": "epsg:9999"}}
```

This yields: `{'vmin': 0, 'vmax': 100, 'geo_config': {'init': 'epsg:9999'}}`

```python

# Mock
self = Survey
self.config = {"geospatial": {"init": "epsg:4326"}}
parameters = {"vmin": 0, "vmax": 100}
```

This yields: `{'vmin': 0, 'vmax': 100, 'geo_config': {'init': 'epsg:4326'}}`

So this is the intended interaction, although I think it may be redundant since the `PlotModel` `pydantic` model will provide a default "EPSG" projection value if none is supplied. This therefore may be unnecessary. 

> the if check for parameters.get("geo_config") is confusing: when parameters["geo_config"] does not exist, the statement geo_config.update(parameters.get("geo_config", None)) will not be executed, but in there you have in there to update the dict geo_config with None. I am not following what you are trying to achieve here, but I think there's some redundancy here that can be removed.

This is related to my last point above re: this check potentially being redundant and therefore unnecessary. The logic is:

1. Did the user provide a geo-configuration in `plot_parameters`? If so, update the stored geo-configuration from the `Survey` object. 
2. Regardless of the presence for `geo_config` in `plot_parameters`, whatever geo-configuration stored within the `Survey` object and those supplied in `plot_parameters` are combined. 

With that in mind, that is a goofy way to approach what I was trying to accomplish here, so I agree that there is some circular dog-chasing-tail behavior going on here that can be clipped.